Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemScriptingHandler.cpp

```diff
@@ -20,6 +20,44 @@
 
 namespace AzToolsFramework::Prefab
 {
+    static AZStd::unique_ptr<AZ::Entity> CreateContainerEntityAndParentEntities(
+        const AZStd::vector<AZ::Entity*> entities, const AZStd::string& filePath)
+    {
+        bool result = false;
+        [[maybe_unused]] AZ::EntityId commonRoot;
+        EntityList topLevelEntities;
+        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+            result, &AzToolsFramework::ToolsApplicationRequestBus::Events::FindCommonRootInactive, entities, commonRoot, &topLevelEntities);
+
+        auto containerEntity = AZStd::make_unique<AZ::Entity>();
+
+        containerEntity->CreateComponent<Components::EditorLockComponent>();
+        containerEntity->CreateComponent<Components::EditorVisibilityComponent>();
+        containerEntity->CreateComponent<Prefab::EditorPrefabComponent>();
+
+        {
+            auto transformComponent = containerEntity->CreateComponent<Components::TransformComponent>();
+            // Because procedural prefabs need to be deterministic we need to set the component ID to something unique and non-random
+            // The prefab that references the proc prefab will store a patch that references the transform component by it's component ID
+            // If this ID is not stable, the proc prefab will lose its position, parenting, etc data next time it is regenerated
+            auto hash = TypeHash64(reinterpret_cast<const uint8_t*>(filePath.data()), filePath.length(), AZ::HashValue64{ 0 });
+
+            transformComponent->SetId(static_cast<AZ::ComponentId>(hash));
+        }
+
+        for (AZ::Entity* entity : topLevelEntities)
+        {
+            AzToolsFramework::Components::TransformComponent* transformComponent =
+                entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+
+            if (transformComponent)
+            {
+                transformComponent->SetParent(containerEntity->GetId());
+            }
```

Comment:
nit: I think this can be simplified. The 'AzToolsFramework' namespace can be omitted.

```cpp
if (auto transformComponent = entity->FindComponent<Components::TransformComponent>())
{
    transformComponent->SetParent(containerEntity->GetId());
}
```